eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServerV2.java

```diff
@@ -0,0 +1,170 @@
+                                                      consequenceType,
+                                              @RequestParam(name = "maf", required = false) String maf,
+                                              @RequestParam(name = "polyphen", required = false) String polyphenScore,
+                                              @RequestParam(name = "sift", required = false) String siftScore,
+                                              @RequestParam(name = "annot-vep-version", required = false) String
+                                                      annotationVepVersion,
+                                              @RequestParam(name = "annot-vep-cache-version", required = false) String
+                                                      annotationVepCacheVersion,
+                                              HttpServletResponse response,
+                                              @ApiIgnore HttpServletRequest request)
+            throws IllegalArgumentException {
+        checkParameters(annotationVepVersion, annotationVepCacheVersion, species);
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" + assembly));
+
+        List<VariantRepositoryFilter> filters = new FilterBuilder()
+                .getVariantEntityRepositoryFilters(maf, polyphenScore, siftScore, studies, consequenceType);
+
+        List<Region> regions = Region.parseRegions(regionId);
+        List<String> excludeMapped = getExcludeMapped();
+
+        AnnotationMetadata annotationMetadata = getAnnotationMetadataHelper(annotationVepVersion,
+                annotationVepCacheVersion);
+        Integer pageSize = service.countByRegionsAndComplexFilters(regions, filters).intValue();
+
+        List<VariantWithSamplesAndAnnotation> variantEntities;
+
+        if (pageSize == 0) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return new ResponseEntity(Collections.EMPTY_LIST, HttpStatus.NOT_FOUND);
+        }
+        try {
+            variantEntities = service.findByRegionsAndComplexFilters(regions,
+                    filters,
+                    annotationMetadata,
+                    excludeMapped,
+                    new PageRequest(0, pageSize));
+        } catch (AnnotationMetadataNotFoundException ex) {
+            return new ResponseEntity(ex.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+        List<Resource> resourcesList = new ArrayList<>();
+        variantEntities.forEach(variantEntity -> {
+            Variant variant = new Variant(variantEntity.getChromosome(), variantEntity.getStart(),
+                    variantEntity.getEnd(), variantEntity.getReference(), variantEntity.getAlternate());
+            variant.setIds(variantEntity.getIds());
+            variant.setMainId(variantEntity.getMainId());
+            Link variantLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getCoreInfo(
```

Comment:
I shouldn't have suggested this link, because the core variant information is already returned by this endpoint. The links should be the same as in the 'variants' endpoint.

Reply:
There is a small issue when multiple resources has to be returned. I have posted this issue on slack :+1:

Reply:
Fixed :)